lightning/restore/restore.go

```diff
@@ -474,6 +472,66 @@ func (rc *RestoreController) runPeriodicActions(ctx context.Context, stop <-chan
 	}
 }
 
+type gcLifeTimeManager struct {
+	runningJobsLock *sync.Mutex
+	runningJobs     *int32
+	oriGCLifeTime   *string
```

Comment:
Instead of a struct of 3 pointers, use a pointer to a struct of 3 values instead.

```go
type gcLifeTimeManager struct {
    runningJobsLock sync.Mutex
    runningJobs     int   // no need to use int32 here, btw.
    oriGCLifeTime   string
}
```